Public/Sdk/Public/Transformers/Transformer.Execute.dsc

```diff
@@ -161,6 +161,14 @@ namespace Transformer {
          * for a pip. The pip will still be stored to the cache for sake of supporting distribution.
          */
         disableCacheLookup?: boolean;
+
+        /**
+         * The weighted value of this pip when limiting concurrency of process pips.
+         * The default weight for a process pip is 1, and the max valid weight is the max concurrent processes allowed in the build.

```

Comment:
>s 1, [](start = 49, length = 4)

should we make the default weight 100?

That would allow for us to fold the 'lightweight' pips in this same concept?

Reply:
We can of course then for the 'slots' to reserve since that probably will have to remain whole numbers.... `Math.max (weight /100)`

---
In reply to: [271089886](https://github.com/Microsoft/BuildXL/pull/67#discussion_r271089886) [](ancestors = 271089886)

Reply:
@semihokur to comment on that

Reply:
I don't know much about our lightweight pips feature, and it is relatively difficult to find in the codebase and is omitted from our wiki.

Do you have any pointers to a description/the implementation of lightweight?

---
In reply to: [271090152](https://github.com/Microsoft/BuildXL/pull/67#discussion_r271090152) [](ancestors = 271090152,271089886)

Reply:
Do you mean comment on what is the "max concurrent processes" allowed in the build?

The max valid value here is technically max int.

I can add a more detailed description here, but probably not so detailed as to actually explain what max concurrent processes is throughout the build, as that would be like full paragraph at least. 

---
In reply to: [271090887](https://github.com/Microsoft/BuildXL/pull/67#discussion_r271090887) [](ancestors = 271090887)

Reply:
They are mostly used for Drop & IPC pips. They have their own lightweight queue in the schedule at the moment. I know we ahve an `IsLight` in the graph (on edges) as well as `IsLight` on the optoins., I'd need to read up on the specific implementation as well. @almilli: Could you chime in if it is worth reducing the concept count, or if they are really different concneps...

---
In reply to: [271091814](https://github.com/Microsoft/BuildXL/pull/67#discussion_r271091814) [](ancestors = 271091814,271090152,271089886)

Reply:
I meant to tag semih regarding potentially folding this into the lightweight pips concept. My comment didn't get associated with the right chain

---
In reply to: [271092886](https://github.com/Microsoft/BuildXL/pull/67#discussion_r271092886) [](ancestors = 271092886,271090887)

Reply:
Perhaps the value of 0 could be used to indicate lightweight processes. 

---
In reply to: [271094619](https://github.com/Microsoft/BuildXL/pull/67#discussion_r271094619) [](ancestors = 271094619,271091814,271090152,271089886)

Reply:
I don't think we still use lightweight pips anywhere.

They were originally introduced before we had IPC pips, and when all our drop pips (those are the pips that talk to DropDaemon) were running as regular processes.  Since those pips just sit around and do nothing while waiting to hear back from DropDaemon, we wanted to able to schedule a whole bunch of them at a time.

Now all drop pips run as IPC pips, which get their own queue and their own default limits/priorities, which makes them completely independent of the settings used for process pips.

Danny's idea makes perfect sense to me.

Reply:
I think it's actually more complicated than that. Semih is tracking an issue where a concurrency change to the light pips introduced a perf regression. Please sync up with him before modifying the meaning of lightweight pips.

---
In reply to: [271102594](https://github.com/Microsoft/BuildXL/pull/67#discussion_r271102594) [](ancestors = 271102594)

Reply:
That might work. Note I don't want to hold the PR on this. Just trying to go through the thought process to avoid multiple concepts trying to do the same thing :)

---
In reply to: [271100050](https://github.com/Microsoft/BuildXL/pull/67#discussion_r271100050) [](ancestors = 271100050,271094619,271091814,271090152,271089886)

Reply:
I am not going to account for lightweight pips in this PR, but I think the scheme of having weight represent # of process slots a pip desires, and having it default to 1 should leave us enough flexibility to do lightweight pips in the future.

We could do some negative scheme like Aleks suggested in his other comment, or just use the value of 0 as the actual value of 0, which means there will always be enough process slots to run the pip, since available process slots is never < 1. 

---
In reply to: [271470356](https://github.com/Microsoft/BuildXL/pull/67#discussion_r271470356) [](ancestors = 271470356,271100050,271094619,271091814,271090152,271089886)

Reply:
If the question is whether light pips should be integrated into the weight concept, I'd like to keep them separate for now until Office gets rid of light esrp pips. 

---
In reply to: [271098357](https://github.com/Microsoft/BuildXL/pull/67#discussion_r271098357) [](ancestors = 271098357,271092886,271090887)

Reply:
We should first wait for Office to stop using light process pips for wrong purposes. It is good to separate them for now. 

---
In reply to: [271105044](https://github.com/Microsoft/BuildXL/pull/67#discussion_r271105044) [](ancestors = 271105044,271102594)

Reply:
Thanks @AnnieFu & @Seokur Sounds good to me.

---
In reply to: [271491641](https://github.com/Microsoft/BuildXL/pull/67#discussion_r271491641) [](ancestors = 271491641,271098357,271092886,271090887)